tests/functional/object/mcg/test_internal_route_auth.py

```diff
@@ -15,13 +22,84 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.fixture(autouse=True)
+def reset_configmap(request):
+    """
+    resets noobaa metrics configmap values to original values
+    """
+    noobaa_configmap = OCP(
+        kind="configmap",
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+        resource_name=constants.NOOBAA_CONFIGMAP,
+    )
+
+    def reset_flag():
+        params = '{{"data": {{"NOOBAA_VERSION_AUTH_ENABLED": "true"}}}}'
+        noobaa_configmap.patch(params=params, format_type="merge")
+        params = '{{"data": {{"NOOBAA_METRICS_AUTH_ENABLED": "true"}}}}'
+        noobaa_configmap.patch(params=params, format_type="merge")
+
+    request.addfinalizer(reset_flag)
+
+
 @mcg
 @red_squad
 class TestNoobaaRouteAuthentication:
     """
     Tests noobaa management route and noobaa endpoints with JWT authentication
     """
 
+    def change_config_flag(self, flag_param):
+        """
+        Changes flag value to false for testing purpose
+        """
+        noobaa_configmap = OCP(
+            kind="configmap",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name=constants.NOOBAA_CONFIGMAP,
+        )
+        params = f'{{"data": {{"{flag_param}": "false"}}}}'
+        noobaa_configmap.patch(params=params, format_type="merge")
+
+    def get_endpoint_host_ip_port(self):
+        """
+        returns endpoint host ip and node port associated with it
+        """
+        noobaa_endpoint_pod = get_noobaa_endpoint_pods()[0]
+        endpoint_host_ip = noobaa_endpoint_pod.get().get("status")["hostIP"]
+        logger.info(endpoint_host_ip)
+        s3_service_data = OCP(
+            kind="service",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name="s3",
+        ).get()
+        all_port_info = s3_service_data["spec"]["ports"]
+        port = None
+        for port_info in all_port_info:
+            if port_info.get("name") == "metrics":
+                port = port_info.get("nodePort")
+        assert port is not None, "Metrics Port info not found in s3 service"
+        return f"http://{endpoint_host_ip}:{port}"
+
+    def get_mgmt_route_address(self):
+        """
+        returns noobaa management route address
+        """
+        s3_route_data = OCP(
+            kind="route",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name="noobaa-mgmt",
+        ).get()
+        return f"https://{s3_route_data['spec']['host']}/version"
+
+    def acccess_url(self, url, header=None, verify=False):
+        """
+        Executes curl request to the given url with or without header and returns status code
+        """
+        response = requests.get(url, headers=header, verify=verify)
+        logger.info(response)
+        return response.status_code
```

Comment:
1. Note the typo in the function name: 3 c's in access instead of 2.
2. Please use this more detailed docstring to clarify usage and return value. I was specifically interested in whether the returned code is a string of an int because of the way this function is used in line 247.
```suggestion
def access_url(self, url, header=None, verify=False):
    """
    Executes curl request to the given url with or without header.

    Args:
        url (str): The target URL to request
        header (dict, optional): A dictionary of HTTP headers to include in the request.
            Defaults to None.
        verify (bool, optional): Whether to verify the server's TLS/SSL certificate.
            Defaults to False.

    Returns:
        int(?): The HTTP status code from the server response.
    """
    response = requests.get(url, headers=header, verify=verify)
    logger.info(response)
    return response.status_code
```

Reply:
Fixed